examples/systemcalls/systemcalls.c

```diff
@@ -1,99 +1,213 @@
     {
         command[i] = va_arg(args, char *);
     }
     command[count] = NULL;
     // this line is to avoid a compile warning before your implementation is complete
     // and may be removed
     command[count] = command[count];
+    initiate_syslog();
+    //attempt to create a child process
+    pid_t process_id = fork();
+    int status;
+    //error handling for fork
+    if (process_id == -1)
+    {
+        syslog(LOG_ERR, "The child process could not be created!");
+        closelog();
+        va_end(args);
+        return false;
+    }
+    else if (process_id == 0)
+    {
+        //replace current process with command[0]
+        if (execv(command[0], command) == -1)
+        {
+            syslog(LOG_ERR, "execv was not successfull");
+            closelog();
+            va_end(args);
+            exit(1); //indicate failure status
+        } 
+    }
 
-/*
- * TODO:
- *   Execute a system command by calling fork, execv(),
- *   and wait instead of system (see LSP page 161).
- *   Use the command[0] as the full path to the command to execute
- *   (first argument to execv), and use the remaining arguments
- *   as second argument to the execv() command.
- *
-*/
+    if (waitpid(process_id, &status, 0) == -1)
+    {
+        syslog(LOG_ERR, "waitpid was not successfull");
+        closelog();
+        va_end(args);
+        return false;
+    }
 
-    va_end(args);
+    if (WIFEXITED(status) && WEXITSTATUS(status) == 0)
+    {
+        syslog(LOG_DEBUG, "child process exited succesfully");
+        closelog();
+        va_end(args);
+        return true;
+    }
 
-    return true;
+    else{
+        syslog(LOG_ERR, "child process exit failed with err code %d",status);
+        closelog();
+        va_end(args);
+        return false;
+    }
 }
 
 /**
-* @param outputfile - The full path to the file to write with command output.
-*   This file will be closed at completion of the function call.
-* All other parameters, see do_exec above
-*/
+ * @param outputfile - The full path to the file to write with command output.
+ *   This file will be closed at completion of the function call.
+ * All other parameters, see do_exec above
+ */
 bool do_exec_redirect(const char *outputfile, int count, ...)
 {
     va_list args;
     va_start(args, count);
-    char * command[count+1];
+    char *command[count + 1];
     int i;
-    for(i=0; i<count; i++)
+    for (i = 0; i < count; i++)
     {
         command[i] = va_arg(args, char *);
     }
     command[count] = NULL;
     // this line is to avoid a compile warning before your implementation is complete
     // and may be removed
     command[count] = command[count];
+    initiate_syslog();
+    int fd = open(outputfile, O_WRONLY | O_TRUNC | O_CREAT, S_IWUSR | S_IRUSR);
+    if (fd == -1)
+    {
+        syslog(LOG_ERR, "Error opening file");
+        closelog();
+        va_end(args);
+        close(fd);
+        return false;
+    }
+    pid_t process_id = fork();
```

Comment:
@nasa7792 think about how you could avoid duplicating logic with the function above.